packages/core/src/common/message-service-protocol.ts

```diff
@@ -53,7 +67,17 @@ export class MessageClient {
         this.logger.info(message.text);
         return Promise.resolve(undefined);
     }
-}
+
+    /**
+     * Create progress message instance.
+     * If a progress message with given text is already shown, returns the shown instance.
+     *
+     * To be implemented by an extension, e.g. by the messages extension.
+     */
+    getOrCreateProgressMessage(message: ProgressMessageArguments): ProgressMessage | undefined {
```

Comment:
1. methods to be callable via jsonrpc need to return a promise, cf. other methods of this service or other services.
2. `MessageService` should be also useable from BE via jsonrpc; therefore it's not possible to send params/result objects with methods/events. it sould be simple (hierarchical) objects only.
3. I believe it's not sufficient to identify progress items by message text. I'd prefer a token/id for that.